source/FreeRTOS_Routing.c

```diff
@@ -615,84 +627,74 @@ struct xIPv6_Couple
-            }
+                ( void ) memset( pxEndPoint, 0, sizeof( *pxEndPoint ) );
 
-            if( pxNetPrefix != NULL )
-            {
-                ( void ) memcpy( pxEndPoint->ipv6_settings.xPrefix.ucBytes, pxNetPrefix->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
-            }
+                pxEndPoint->bits.bIPv6 = pdTRUE_UNSIGNED;
 
-            /* Copy the current values to the default values. */
-            ( void ) memcpy( &( pxEndPoint->ipv6_defaults ), &( pxEndPoint->ipv6_settings ), sizeof( pxEndPoint->ipv6_defaults ) );
+                pxEndPoint->ipv6_settings.uxPrefixLength = uxPrefixLength;
 
-            ( void ) memcpy( pxEndPoint->ipv6_defaults.xIPAddress.ucBytes, pxIPAddress->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+                if( pxGatewayAddress != NULL )
+                {
+                    ( void ) memcpy( pxEndPoint->ipv6_settings.xGatewayAddress.ucBytes, pxGatewayAddress->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+                }
 
-            #if ( ipconfigUSE_DHCPv6 != 0 ) || ( ipconfigUSE_RA != 0 )
-                xIsStatic = pdTRUE;
-                #if ( ipconfigUSE_DHCPv6 != 0 )
-                    if( pxEndPoint->bits.bWantDHCP != 0U )
-                    {
-                        xIsStatic = pdFALSE;
-                    }
-                #endif
-                #if ( ipconfigUSE_RA != 0 )
-                    if( pxEndPoint->bits.bWantRA != 0U )
-                    {
-                        xIsStatic = pdFALSE;
-                    }
-                #endif
+                if( pxDNSServerAddress != NULL )
+                {
+                    ( void ) memcpy( pxEndPoint->ipv6_settings.xDNSServerAddresses[ 0 ].ucBytes, pxDNSServerAddress->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+                }
 
-                if( xIsStatic == pdTRUE )
-            #endif /* ( ipconfigUSE_DHCPv6 != 0 ) || ( ipconfigUSE_RA != 0 ) */
-            {
-                ( void ) memcpy( pxEndPoint->ipv6_settings.xIPAddress.ucBytes, pxIPAddress->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
-            }
+                if( pxNetPrefix != NULL )
+                {
+                    ( void ) memcpy( pxEndPoint->ipv6_settings.xPrefix.ucBytes, pxNetPrefix->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+                }
 
-            ( void ) memcpy( pxEndPoint->xMACAddress.ucBytes, ucMACAddress, ipMAC_ADDRESS_LENGTH_BYTES );
-            ( void ) FreeRTOS_AddEndPoint( pxNetworkInterface, pxEndPoint );
+                /* Copy the current values to the default values. */
+                ( void ) memcpy( &( pxEndPoint->ipv6_defaults ), &( pxEndPoint->ipv6_settings ), sizeof( pxEndPoint->ipv6_defaults ) );
+
+                ( void ) memcpy( pxEndPoint->ipv6_defaults.xIPAddress.ucBytes, pxIPAddress->ucBytes, ipSIZE_OF_IPv6_ADDRESS );
+
+                ( void ) memcpy( pxEndPoint->xMACAddress.ucBytes, ucMACAddress, ipMAC_ADDRESS_LENGTH_BYTES );
```

Comment:
Why are we not setting the static IP address here, shouldn't we set the endpoint IP address as the static IP address provided by the user if neither DHCPv6 nor RA is enabled? If we expect DHCP or RA to giveup and then assign the default IP we might have to wait till the timeout for the endpoint to be up. Also DHCP or RA can be disabled during build.

Reply:
That's a good point. Please allow me to list down my answer below:
1. We have a [reset on endpoint](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/845/files/c80762bb530940fda9c9fc58ed1c0647059ae601#diff-cabc7b626b59843c03a1b0a4018200289c01a46d67b68e3445249b18db94a853R644) in this function, so the DHCP/RA bits are always zero. It's useless to check DHCP/RA before setting IP.
2. We use a general way to set IP addresses for endpoints, it's in [prvProcessNetworkDownEvent](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/c80762bb530940fda9c9fc58ed1c0647059ae601/source/FreeRTOS_IP_Utils.c#L896-L906) for now. It may make flow more complicated if we can't tell if an endpoint is for DHCP/RA or not.

Reply:
Yes you're right, @ActoryOu, `memset()` is called to clear the entire endpoint.

I added this code for a reason:
Endpoint 1 (global IPv6) needs DHCPv6
Endpoint 2 (static local-link) is used to communicate with the DHCP server

I would like to set 'ipv6_settings.xIPAddress' for all static endpoints, before `prvProcessNetworkDownEvent()` is called for the first time. Otherwise DHCP can not find a local-link IPv6 address.

Reply:
Hi @htibosch,
Thanks for comment.
But one query, how do we know if the endpoint is static in this function?
Or is it ok to always set ipv6_settings/ipv4_settings IP address for endpoints?